packages/klingon-ui/src/app/app.component.ts

```diff
@@ -60,22 +63,51 @@ export class SnackBarSuccessComponent {}
 export class AppComponent implements OnInit {
   selectedIndex = 0;
 
+  // Enable/Disable generate tab
+  klingon = { generate: false };
+
   @ViewChild('appCli') appCli: CliCreateComponent;
+  @ViewChild('appGenerate') appGenerate: CliGenerateComponent;
 
   constructor(
     public snackBarError: MatSnackBar,
-    public snackBarSuccess: MatSnackBar
-  ) {}
+    public snackBarSuccess: MatSnackBar,
+    private cliService: CliService
+  ) { }
 
   ngOnInit() {
     this.selectedIndex = parseInt(
       localStorage.getItem('ui.selectedIndex') || '0',
       10
     );
     localStorage.setItem('ui.selectedIndex', `${this.selectedIndex}`);
+
+    this.subscribeToNgCreate();
+  }
+
+  subscribeToNgCreate() {
+    this.cliService.onNgCreate.asObservable()
+      .pipe(flatMap((data: Subject<CommandResult>) => data))
+      .subscribe((response: any) => {
+        /**
+        * exit event of ng command returns exit code (0/1). So if it returns 0, means ng command executed successfully. Only then
+        * we enable generate tab.
+        */
+        if (response.exit === 0) {
```

Comment:
I think we should not switch to the generate UI if we are in the Dry-Run mode.

Reply:
Done . @manekinekko